app/models/concerns/resource_accessibility.rb

```diff
@@ -0,0 +1,24 @@
+module ResourceAccessibility
```

Comment:
Instead of moving this logic to a concern, you can pass the view context to the pagelet `onlyif` Proc by passing `self` as an extra parameter when calling the proc from PageletHelper. This will allow you to use the existing helpers.